### PR TITLE
Fix Haiku config directory and SDL 1.2 build

### PR DESCRIPTION
**Pull request by Crestwave**

### CODE REVIEW

File: src/core/config.c [hunk 1]
```diff
@@ -227,6 +231,12 @@ void mCoreConfigDirectory(char* out, size_t outLength) {
 	UNUSED(portable);
 	snprintf(out, outLength, "/%s", projectName);
 	FSUSER_CreateDirectory(sdmcArchive, fsMakePath(PATH_ASCII, out), 0);
+#elif defined(__HAIKU__)
+	UNUSED(portable);
```
Comment (endrift) [inline src/core/config.c]: You should probably have a portable.ini check like the other desktop OSes do.

Comment (Crestwave) [reply src/core/config.c]: Oh, I thought that it was for a portable build distributed for specific OSes or something. I haven't tested it yet but I'm guessing a copy of the other check will do

Comment (Crestwave) [reply src/core/config.c]: Okay, so I built it and launched it in a directory with a file named portable.ini and the config files were placed in that directory. That's the expected behavior, yes?

Comment (endrift) [reply src/core/config.c]: Yes, that's expected behavior.

**Pull request merged**